llumnix/backends/vllm/worker.py

```diff
@@ -66,135 +84,63 @@ def init_migration(self, num_migration_cache_blocks: int, src_worker_handle_list
-
-    def send_cpu_cache(self, blocks: List[int]):
-        num_blocks = len(blocks)
-        dummy_key_cpu = self.migration_key_cache[:num_blocks]
-        dummy_value_cpu = self.migration_value_cache[:num_blocks]
-        with torch.cuda.stream(self.migration_stream):
-            for layer_idx in range(self.cache_engine.num_layers):
-                for idx, block_num in enumerate(blocks):
-                    dummy_key_cpu[idx][layer_idx].copy_(self.gpu_cache[layer_idx][0][block_num])
-                    dummy_value_cpu[idx][layer_idx].copy_(self.gpu_cache[layer_idx][1][block_num])
-        torch.cuda.Stream.synchronize(self.migration_stream)
-        return (dummy_key_cpu.to(self.rpc_dtype).numpy(), dummy_value_cpu.to(self.rpc_dtype).numpy())
-
-    def recv_cpu_cache(self, blocks: List[int], rpc_numpy_cache):
-        num_blocks = len(blocks)
-        dummy_key = self.migration_key_cache[:num_blocks]
-        dummy_value = self.migration_value_cache[:num_blocks]
-        k = rpc_numpy_cache[0]
-        v = rpc_numpy_cache[1]
-        dummy_key.copy_(torch.from_numpy(k))
-        dummy_value.copy_(torch.from_numpy(v))
-        with torch.cuda.stream(self.migration_stream):
-            for layer_idx in range(self.cache_engine.num_layers):
-                for idx, block_num in enumerate(blocks):
-                    self.gpu_cache[layer_idx][0][block_num].copy_(dummy_key[idx][layer_idx])
-                    self.gpu_cache[layer_idx][1][block_num].copy_(dummy_value[idx][layer_idx])
-        torch.cuda.Stream.synchronize(self.migration_stream)
-
-    def send_cpu_cache_v2(self, blocks: List[int]):
-        src_to_dst = dict(enumerate(blocks))
-        with torch.cuda.stream(self.migration_stream):
-            for layer_idx in range(self.cache_engine.num_layers):
-                self.cache_engine.attn_backend.swap_blocks(self.gpu_cache[layer_idx], self.migration_cache[layer_idx], src_to_dst)
-        torch.cuda.Stream.synchronize(self.migration_stream)
-        return self.migration_cache.to(self.rpc_dtype).numpy()
-
-    def recv_cpu_cache_v2(self, blocks: List[int], rpc_numpy_cache):
-        with torch.cuda.stream(self.migration_stream):
-            self.migration_cache.copy_(torch.from_numpy(rpc_numpy_cache))
-            src_to_dst = dict(enumerate(blocks))
-            for layer_idx in range(self.cache_engine.num_layers):
-                self.cache_engine.attn_backend.swap_blocks(self.migration_cache[layer_idx], self.gpu_cache[layer_idx],src_to_dst)
-        torch.cuda.Stream.synchronize(self.migration_stream)
-
-
-    def migrate_gpu_cache_ray_rpc(self, src_worker_handle_list, src_blocks: List[int], dst_blocks: List[int]):
+
+        self.instance_id = instance_id
+        self.migration_config = migration_config
+        num_instance = len(migration_config.instance_rank_map)
+        self.ray_world_size = num_instance * self.parallel_config.world_size
```

Comment:
rank_world_size -> global_world_size, so is ray_rank, because world_size and rank here have nothing to do with ray.